unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/QuestsTrackerHUD/QuestsTrackerTask.cs

```diff
@@ -47,13 +63,33 @@ public void Populate(QuestTask task)
             }
         }
 
-        private void Update()
+        public IEnumerator ProgressAndCompleteSequence()
         {
             Vector3 scale = progress.transform.localScale;
-            if (Math.Abs(scale.x - progressTarget) < Mathf.Epsilon)
-                return;
-            scale.x = Mathf.MoveTowards(scale.x, progressTarget, 0.1f);
-            progress.transform.localScale = scale;
+            if (Math.Abs(scale.x - progressTarget) > Mathf.Epsilon)
+                progressBarIncrementAudioEvent.Play();
+            while (scale.x < progressTarget)
+            {
+                scale.x = Mathf.MoveTowards(scale.x, progressTarget, Time.deltaTime);
+                progress.transform.localScale = scale;
+                yield return null;
+            }
+            if (progressTarget < 1)
+                yield break;
+
+            //Dont play completed animation if is already playing
+            if (animator.GetCurrentAnimatorClipInfo(0).All(x => x.clip.name != "QuestTrackerTaskCompleted"))
```

Comment:
Can we have this "QuestTrackerTaskCompleted" in a const?

Reply:
Done!

Reply:
Done!